### PR TITLE
Fixes Queens being able to gib marines with huggers attached

### PR DESCRIPTION
**Pull request by realforest2001**
## About The Pull Request
Fixes Queen Gib to check for larva both before initiation and after, to prevent hugged humans being gibbed after infection when started before.

## Changelog

:cl:
fix: Queen Gib no longer works on hugged marines pre-infection
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/castes/Queen.dm [hunk 1]
```diff
@@ -640,6 +640,36 @@
...
+	if(isXeno(victim))
+		var/mob/living/carbon/Xenomorph/xeno = victim
+		if(hivenumber == xeno.hivenumber)
+			to_chat(src, SPAN_WARNING("You can't bring yourself to harm a fellow sister to this magnitude."))
+			return FALSE
+
+	var/mob/living/carbon/human/H = victim
+	if(istype(H))
+		if(istype(H.wear_mask, /obj/item/clothing/mask/facehugger))
+			to_chat(src, SPAN_XENOWARNING("The host will soon bare a child of the hive!"))
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/castes/Queen.dm:658]:
```suggestion
			to_chat(src, SPAN_XENOWARNING("The host will soon bear a child of the hive!"))
```

**Review by stanalbatross — CHANGES_REQUESTED** [review]

**Comment by stanalbatross** [comment]
image
image

if you could, can you also fix this crash when gibbing an alive person (it also makes the death message not show up in dchat)

File: code/modules/mob/living/carbon/xenomorph/castes/Queen.dm [hunk 2]
```diff
@@ -640,6 +640,36 @@
...
+			return FALSE
+
+		if(locate(/obj/item/alien_embryo) in victim) //Maybe they ate it??
+			if(H.status_flags & XENO_HOST)
+				if(victim.stat != DEAD) //Not dead yet.
+					to_chat(src, SPAN_XENOWARNING("The host and child are still alive!"))
+					return FALSE
+				else if(( world.time <= H.timeofdeath + H.revive_grace_period )) //Dead, but the host can still hatch, possibly.
+					to_chat(src, SPAN_XENOWARNING("The child may still hatch! Not yet!"))
+					return FALSE
```
Comment (Watermelon914) [inline code/modules/mob/living/carbon/xenomorph/castes/Queen.dm:668]: make H a more detailed variable name

**Comment by Carnage068** [comment]
no offense or anything but when would this even happen in game?

**Comment by stanalbatross** [comment]
image
issue still happening
image
likely a problem with cause_data there

**Comment by realforest2001** [comment]
Sigh. No idea why it keeps freaking out, this cause_data stuff is a pain.

**Comment by realforest2001** [comment]
> no offense or anything but when would this even happen in game?

I've witnessed it more than once, which is enough reason to fix it. Even if it never happened, no reason to make sure it's checking things properly anyway.

**Comment by stanalbatross** [comment]
> Sigh. No idea why it keeps freaking out, this cause_data stuff is a pain.

mostly because it's string data rather than datums as you have create_cause_data("xyz") all over there

**Comment by Watermelon914** [comment]
keep your title names descriptive please

File: code/modules/mob/living/carbon/xenomorph/castes/Queen.dm [hunk 3]
```diff
@@ -640,6 +640,36 @@
...
+			return FALSE
+
+	if(isXeno(victim))
+		var/mob/living/carbon/Xenomorph/xeno = victim
+		if(hivenumber == xeno.hivenumber)
+			to_chat(src, SPAN_WARNING("You can't bring yourself to harm a fellow sister to this magnitude."))
+			return FALSE
+
+	var/mob/living/carbon/human/Host = victim
+	if(istype(Host))
```
Comment (Watermelon914) [inline code/modules/mob/living/carbon/xenomorph/castes/Queen.dm:656]: use snake_case for variable names, avoid PascalCase

**Pull request closed without merging**